### DISCUSSION TITLE
Supports internal field keys from relational data in 'count' queries.

### DISCUSSION THREAD

**Discussion by g2ilgu** [prisma/prisma#21816] [Q&A] [answered]
## Problem
I need a `count` query to find the number of data with matching internal fields in relational data.

Can I replace the raw query below with a function from prisma?
Below is the raw query I implemented.
```
const query = Prisma.sql`
      SELECT
        COUNT(*) ::INT AS organization_name_matched_comment_count
      FROM
        comment AS co
        INNER JOIN post AS po ON co.post_id = po.id AND co.organization_name = po.organization_name
      WHERE
        co.parent_id IS NULL
        AND co.created_at >= ${startDate}
        AND co.created_at <= ${endDate};
    `
const [
      {
        organization_name_matched_comment_count:
          organizationNameMatchedCommentCount = 0,
      },
    ] = await this.prisma.$queryRaw<
      [{ organization_name_matched_comment_count: number }]
    >(query)
```

### prisma example
```
model Post {
  id                   BigInt    @id
  title                String
  organizationName     String   @map("organization_name")
  createdAt            DateTime  @default(now()) @map("created_at") @db.Timestamp(3)
  deletedAt            DateTime? @map("deleted_at") @db.Timestamp(3)

  comments         Comment[]

  @@map("post")
}

model Comment {
  id                      BigInt    @id
  parentId                BigInt?   @map("parent_id")
  postId                  BigInt    @map("post_id")
  content                 String    @db.Text
  organizationName        String   @map("organization_name")
  createdAt               DateTime  @default(now()) @map("created_at") @db.Timestamp(3)
  updatedAt               DateTime? @map("updated_at") @db.Timestamp(3)
  deletedAt               DateTime? @map("deleted_at") @db.Timestamp(3)

  post Post  @relation(fields: [postId], references: [id])

  @@index([postId])
  @@map("comment")
}
```

## Suggested solution

Sequelize supports the `sequence.col('table.column')` function. You may need the following features:

## Alternatives

Use the raw query in the example above.

## Additional context

OS: macOS
database: PostgrSQL
Node.js version: 18.16.0

**Comment by millsp** [comment]
Hey @G2ilgu, I think you could for instance encapsulate this specific method int a model [extension](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions). Alternatively, did I understand correctly that you are looking for a type-safe query builder?

**Reply by g2ilgu** [reply]
@millsp Thank you for your answer. I'm looking for a type-safe query builder as you said. Is there a way to replace the raw query above in prisma?

**Reply by millsp** [reply, marked as answer]
Thanks. Then you can upvote this issue https://github.com/prisma/prisma/issues/13017 so that we can prioritize it faster in the future.
